added hexxat cloak items

bulian-bg opened the pull request.

bulian-bg committed:
Hexxat cloak dialogue

bulian-bg committed:
hexxat dialogue update

bulian-bg committed:
hexxat cloak dialogue

bulian-bg committed:
added hexxat's cloak

bulian-bg committed:
added hexxat items

bulian-bg committed:
add cloak of the dead

bulian-bg committed:
Update setup.tra

bulian-bg committed:
added hexxat's cloak

bulian-bg committed:
add hexxat cloak

bulian-bg committed:
added hexxat cloak

bulian-bg committed:
Update cespy.d

bulian-bg committed:
Update cespy.tra

bulian-bg committed:
Update crom.tra

critto-bg commented on ImprovedAnvil/lang/English/crom.tra:
```diff
@@ -238,3 +238,7 @@
+@640 = ~What can you do with Cloak of Dragomir?~
+@641 = ~Sadly, I can be of no help. Track down a fellow by name of Maheer. An eccentric man, truth be told to ye. But they say he has a knack for handling delicate material such as one that ye have.~
+@642 = ~What can you do with this Enhanced Cloak of Dragomir?~
+@643 = ~Sadly, I can be of no help. Track down a fellow by name of Maheer. An eccentric man, truth be told to ye. But they say he has a knack for handling delicate material such as one that ye have.~
```
This is purely cosmetic, but maybe the second line shouldn't be copied verbatim. Also, if you do make them identical, there's no need to duplicate the string, just reference @641 both times.

bulian-bg replied:
updated the second line which seems preferable to keep the indexing consistent

critto-bg commented on ImprovedAnvil/lang/English/setup.tra:
```diff
@@ -7103,3 +7103,46 @@ Sarevok~
+@12032 = ~Cloak of the Daywalker~
+@12033 = ~Cloak of the Daywalker
+
+Using Adamantite Dust to absorb the magical aura from Dragomir’s Cloak and an additional Cloak of Protection, Dragonmir’s Cloak has reached its full potential. 
+
```
"With the help of..." instead of "Using..."?

critto-bg commented on ImprovedAnvil/lang/English/shop03.tra:
```diff
@@ -16,4 +16,9 @@
+@5109 = ~Let us proceed!~
+
+@5110 = ~Can you fashion something out of this Cloak of Dragomir?~
+@5111 = ~I can line your cloak  with a Cloak of the Dead and a Cloak of Protection +1 for a modest fee of 25,000 gold.~
```
extra space between "cloak" and "with"

critto-bg commented on ImprovedAnvil/resources/dlg/cespy.d:
```diff
@@ -2125,6 +2129,36 @@ IF ~~ THEN BEGIN treefist
+          TakePartyItemNum("s!clck08",1)   DestroyItem("s!clck08")
+          TakePartyItemNum("clck01",1)   DestroyItem("clck01")
+          TakePartyItemNum("dwdust",1)   DestroyItem("dwdust")
+          TakePartyItemNum("s!misc06",1)   DestroyItem("s!cmisc06")	
```
a typo here: "S!cmisc06"

critto-bg commented on ImprovedAnvil/resources/dlg/cespy.d:
```diff
@@ -2125,6 +2129,36 @@ IF ~~ THEN BEGIN treefist
+  IF ~~ THEN BEGIN hexxat2
+    SAY @793 IF ~~ THEN REPLY @2 GOTO 13
+    IF ~PartyHasItem("s!clck08") PartyGoldGT(49999) 
+	PartyHasItem("clck01") PartyHasItem("dwdust") PartyHasItem("s!misc06") ~
```
Maybe use a stronger cloak of protection in the second recipe? Feels too easy for such a powerful item

bulian-bg replied:
At this point in the game, a cloak of protection +2 is just extra gold.  I kept combine cost consistent with other NPC items but we can increase it if you like or change the recipe to +2.  seems like there are more fundamental questions out there first.

critto-bg commented on ImprovedAnvil/resources/dlg/shop03.d:
```diff
@@ -36,4 +38,36 @@ APPEND ~SHOP03~
+          TakePartyItemNum("s!clck08",1)   DestroyItem("s!clck08")
+          TakePartyItemNum("clck01",1)   DestroyItem("clck01")
+          TakePartyItemNum("dwdust",1)   DestroyItem("dwdust")
+          TakePartyItemNum("s!misc06",1)   DestroyItem("s!cmisc06")	
```
same typo here

critto-bg commented on ImprovedAnvil/scripts/encounters/graveyard/S!crypt.baf:
```diff
@@ -8,6 +8,14 @@ THEN
     SetGlobal("Iaqueen","AR0806",1)
 END
 
+IF
```
use two spaces instead of tabulation here, please (you can probably set this up in the text editor as the default behavior)
also, maybe have this thing dropped only when Hexxat is in party?

bulian-bg replied:
prefer not to have conditional drops

critto-bg reviewed (changes requested):
I haven't checked the items themselves. Will do that late as well, to make sure they are set up properly.

critto-bg commented:
One more thing to consider here. Dragomir's cloak is a semi-quest item for Hexxat. It is being referenced as a trigger in certain dialogues, as well as in-game scripts (including the ever-annoying checks for her being affected by the sunlight). A quick search in NI shows these references.

bulian-bg committed:
Update crom.tra

bulian-bg committed:
Update setup.tra

bulian-bg committed:
Update shop03.tra

bulian-bg committed:
Update cespy.d

bulian-bg committed:
Update shop03.d

bulian-bg committed:
Update S!crypt.baf

critto-bg commented:
All changes are OK but see my comments in the google document. Cloak of dragomir is used heavily in Hexxat's scripting. Simply replacing it will break all of her daylight burn logic.

bulian-bg reviewed (commented):
> One more thing to consider here. Dragomir's cloak is a semi-quest item for Hexxat. It is being referenced as a trigger in certain dialogues, as well as in-game scripts (including the ever-annoying checks for her being affected by the sunlight). A quick search in NI shows these references.

The following script should generally take care of most things.  if you can do the dialogue files, i'll do the script files.

OR(3)
	HasItemEquiped("ohhcloak",Myself)  // Cloak of Dragomir
	HasItemEquiped("s!clck08",Myself)  // Cloak of Dragomir
	HasItemEquiped("s!clck09",Myself)  // Cloak of Dragomir

Looking through the scripts - 
HAXXA25.bcs - some finesse, probably best to manually update
hexxat.bcs - some finesse, probably best to manually update
OHHMORTL.BCS unimportant should be modified
OHHPHREY.BCS - need to set state based on Iahexxatforged variable (by the way are nested if statements possible in this language?)

BEDWIN.DLG trigger 9 - or clause
BHEXXA25.DLG trigger 2 - or clause
OHHPHREY.DLG (trigger 2, trigger 5, action 28 need modified

critto-bg commented:
I'd avoid completely rewriting the file. The scripts could be decompiled, followed by replacing the text and then compiling (this is just instructions to weidu, similar to those found in lib/scripts/01_decompile_and_patch.tph if you want to take a look at those yourself. Just take care to differentiate between HasItemEquipped vs. !HasItemEquipped. Each of those should be replaced separately with the respective set of OR(3)'s. It might be tricky, due to how text search works.

Another possibility is to grab all of the blocks from the original file that check for Dragomir's cloak, make similar ones but for the new ones, put it into a new script and extend it on top of the vanilla one.

Let me know if it makes sense.

bulian-bg commented:
> I'd avoid completely rewriting the file. The scripts could be decompiled, followed by replacing the text and then compiling (this is just instructions to weidu, similar to those found in lib/scripts/01_decompile_and_patch.tph if you want to take a look at those yourself. Just take care to differentiate between HasItemEquipped vs. !HasItemEquipped. Each of those should be replaced separately with the respective set of OR(3)'s. It might be tricky, due to how text search works.
> 
> Another possibility is to grab all of the blocks from the original file that check for Dragomir's cloak, make similar ones but for the new ones, put it into a new script and extend it on top of the vanilla one.
> 
> Let me know if it makes sense.



> I'd avoid completely rewriting the file. The scripts could be decompiled, followed by replacing the text and then compiling (this is just instructions to weidu, similar to those found in lib/scripts/01_decompile_and_patch.tph if you want to take a look at those yourself. Just take care to differentiate between HasItemEquipped vs. !HasItemEquipped. Each of those should be replaced separately with the respective set of OR(3)'s. It might be tricky, due to how text search works.
> 
> Another possibility is to grab all of the blocks from the original file that check for Dragomir's cloak, make similar ones but for the new ones, put it into a new script and extend it on top of the vanilla one.
> 
> Let me know if it makes sense.

I'll give the latter attempt a try for the scripts.  If you have bandwidth to tackle the dialogue changes, that would probably be for the best.

critto-bg commented:
I can do all of this if you want, it's not a problem for me. I will be reviewing the changes very carefully no matter who implements them because there's lots of room for error.

bulian-bg commented:
i added the script modifications in the "hexxat scripts" pull request

critto-bg commented:
OK, I will be checking everything out later. If you want to take a stab at the dialogues, the key principles are pretty much the same: there are triggers (which are IF conditions, basically) and actions (which are sort of like THEN action lists). Both could be replaced textually in the same manner as BCS scripts (there are plenty of examples in lib/dialogue/decompile_and_patch.tph).

The other option is to use WeiDU's dialogue manipulation functionality to replace trigger and action blocks, but is more involved and definitely requires some reading of the documentation and re-checking the results carefully in NI as you go.

bulian-bg committed:
patch OHHPHREY.dlg

i'm a little nervous about the last line - hopefully the actionoverride command will work if the target doesn't have the item

bulian-bg commented:
gave it a shot, somewhat concerned about ohhphrey action 28.  let's see.

critto-bg commented:
Looks fine at first glance, will re-check everything later once I'm done with the custom kits

critto-bg commented on ImprovedAnvil/lib/dialogues/02_decompile_and_patch.tph:
```diff
@@ -132,4 +132,22 @@ COPY_EXISTING ~NEERAJ.DLG~ ~override~
+   REPLACE_TEXTUALLY ~!HasItemEquipedReal("ohhcloak","hexxat")~ ~tempvar1~
+   REPLACE_TEXTUALLY ~HasItemEquipedReal("ohhcloak","hexxat")~ ~OR(3) HasItemEquipedReal("ohhcloak","hexxat") HasItemEquipedReal("S!clck08","hexxat") HasItemEquipedReal("S!clck09","hexxat")~
+   REPLACE_TEXTUALLY ~tempvar1~ ~!HasItemEquipedReal("ohhcloak","hexxat") !HasItemEquipedReal("S!clck08","hexxat") !HasItemEquipedReal("S!clck09","hexxat")~
+   REPLACE_TEXTUALLY ~ActionOverride("hexxat",GiveItem("ohhcloak","ohhphrey"))~ ~ActionOverride("hexxat",GiveItem("ohhcloak","ohhphrey")) ActionOverride("hexxat",GiveItem("S!clck08","ohhphrey")) ActionOverride("hexxat",GiveItem("S!clck09","ohhphrey"))~
```
I wonder what happens to GiveItem() if the item does not exist. Hopefully, just nothing and the scripts won't break

critto-bg reviewed (approved):
The rest looks OK, will be trying to put this thing together now.

critto-bg commented:
The items had mistakes in effects (targeting, timing mode, probability). I will fix them locally. Also, why does the cloak of the daywalker has stoneskin magic ability? I'm removing it.

critto-bg commented:
Merged into master.
